core/src/main/java/org/vertexium/query/QueryBase.java

```diff
@@ -351,6 +351,11 @@ public HasValueContainer(final String key, final Predicate predicate, final Obje
             this.value = value;
             this.predicate = predicate;
             this.propertyDefinitions = propertyDefinitions;
+
+            PropertyDefinition propertyDefinition = PropertyDefinition.findPropertyDefinition(propertyDefinitions, key);
+            if (!predicate.isSupported(propertyDefinition)) {
```

Comment:
Instead of `isSupported` should we make this something like `validate(propertyDefinition)` that way we can through a more meaningful error message, for example in the case below we can say you are missing full text hint.

Reply:
@joeferner Good idea! Changed with 1cf5098361e7756397ca838422294ed2866ccc63.